### PR TITLE
Fixed issue where go.property(msg.url("socket", "path", "fragment")) (with 3 arguments) could not be parsed by Bob.

### LINKED ISSUE #9804: Comparison for URL provided in `go.property` behaves differently between debug and release builds.
**Describe the bug (REQUIRED)**

Doing an equality comparison between a URL provided in a `go.property` and one written in code yields different results between debug and release builds.

**To Reproduce (REQUIRED)**

Build the attached project and observe the blob sprite being different colours in debug vs. release builds.

**Expected behavior (REQUIRED)**

The blob should go the same colour in both debug and release builds.

**Defold version (REQUIRED):**
  - Version 1.9.4

**Platforms (REQUIRED):**

 - Platforms: macOS

**Minimal repro case project (OPTIONAL):**

[NullableURL.zip](https://github.com/user-attachments/files/17829449/NullableURL.zip)

@britzl adding script file for convenience:

```lua
go.property("some_url", msg.url("does", "not", "exist"))

function init(self)
	if self.some_url == msg.url("does", "not", "exist") then
		go.set("#sprite", "tint", vmath.vector4(0, 1, 0, 1))
	else
		go.set("#sprite", "tint", vmath.vector4(1, 0, 0, 1))
	end
end
```

### PR DESCRIPTION
**Pull request by AGulev**
Fix https://github.com/defold/defold/issues/9804

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/test_props_url.lua [hunk 1]
```diff
@@ -2,3 +2,7 @@ go.property("prop1", msg.url("url"))
 go.property("prop2", msg.url(""))
 go.property("prop3", msg.url())
 go.property("prop4", msg.url('url'))
+go.property("prop5", msg.url(''))
+go.property("prop6", msg.url('socket', '/path/to/object', 'fragment'))
+go.property("prop7", msg.url(hash('socket-hash'), hash('/path/to/object-hash'), hash('fragment-hash')))
+go.property("prop8", msg.url("", "", ""))
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/test_props_url.lua:8]: Test cases from the editor and more (but see my comment here pls https://github.com/defold/defold/issues/9804#issuecomment-2504513482 )

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java [hunk 2]
```diff
@@ -302,6 +303,42 @@ private String getFirstStringArg(LuaParser.ArgsContext argsCtx) {
         return initialToken.getText().replace(QUOTES.getOrDefault(initialToken.getType(), ""), "");
     }
 
+    private List<String> getAllStringArgs(LuaParser.ArgsContext argsCtx) {
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java:306]: a function that iterates deeper to parse 3 arguments or/and if it's in hashes.

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java [hunk 3]
```diff
@@ -474,27 +511,40 @@ private ParserRuleContext parsePropertyValue(LuaParser.ArgsContext argsCtx, Prop
...
+                        List<String> allArgs = getAllStringArgs(ctx.nameAndArgs().args());
+                        property.value = concatenateUrl(allArgs);
                     }
-                    property.value = firstStrArg == null ? "" : firstStrArg;
                 }
             }
         }
         return resultContext;
     }
+    public static String concatenateUrl(List<String> allArgs) {
```
Comment (AGulev) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java:541]: Url concatinator - easy way to create an url from a list of strings

Comment (JCash) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java:541]: Does this handle, size == 2 ?

**Review by JCash — DISMISSED** [review]
lgtm.
One question about code coverage.

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java (hunk 3 above)
Comment (AGulev) [reply com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/LuaScanner.java]: Fixed, thank you!

**Review by britzl — APPROVED** [review]

**Pull request merged**